### PR TITLE
refactor CRUDControllerTest to use real container instance

### PR DESCRIPTION
**Pull request by dmaicher**
## Subject

Whille checking this deprecation

```
1x: The "Symfony\Bundle\FrameworkBundle\Controller\ControllerTrait" trait is considered internal. It may change without further notice. You should not use it from "Sonata\AdminBundle\Controller\CRUDController".
```

i had a look at the test for the controller.

It was using a ContainerInterface mock that actually does not behave like a real container instance at all. The mock was not adhering to the interface's contract. 

For example a PSR containe (and also the implementation from Symfony) should throw an exception by default if a service or parameter does not exist.

=> **The container mock simply returned null.**

So I don't think it makes sense to mock the container and over-complicate the test. Simply using a real container seems a lot better here.

I am targeting this branch, because its BC.

### CODE REVIEW

File: tests/Controller/CRUDControllerTest.php [hunk 1]
```diff
@@ -538,7 +488,7 @@ public function testConfigureWithException(): void
     public function testConfigureWithException2(): void
     {
         $this->expectException(\InvalidArgumentException::class);
-        $this->expectExceptionMessage('Found service "nonexistent.admin" is not a valid admin service');
+        $this->expectExceptionMessage('You have requested a non-existent service "nonexistent.admin".');
```
Comment (dmaicher) [inline tests/Controller/CRUDControllerTest.php]: this actually shows the issue nicely.

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Admin/Pool.php#L353

A non-existent service will result in this exception.

File: tests/Controller/CRUDControllerTest.php [hunk 2]
```diff
@@ -253,82 +258,27 @@ protected function setUp(): void
...
+        $this->container->set('twig', $twig);
+        $this->container->set('session', $this->session);
+        $this->container->set('sonata.admin.exporter', $exporter);
+        $this->container->set('sonata.admin.audit.manager', $this->auditManager);
+        $this->container->set('sonata.admin.object.manipulator.acl.admin', $this->adminObjectAclManipulator);
+        $this->container->set('security.csrf.token_manager', $this->csrfProvider);
+        $this->container->set('logger', $this->logger);
+        $this->container->set('kernel', $this->kernel);
+        $this->container->set('translator', $this->translator);
+        $this->container->set('sonata.admin.breadcrumbs_builder', new BreadcrumbsBuilder([]));
```
Comment (dmaicher) [inline tests/Controller/CRUDControllerTest.php]: this service was not set at all previously.

File: tests/Controller/CRUDControllerTest.php [hunk 3]
```diff
@@ -153,13 +156,17 @@ class CRUDControllerTest extends TestCase
      */
     private $formBuilder;
 
+    /**
+     * @var LoggerInterface|MockObject
```
Comment (core23) [inline tests/Controller/CRUDControllerTest.php]:
```suggestion
     * @var LoggerInterface&MockObject
```

**Comment by greg0ire** [comment]
Since this won't bring any benefit to the end user, I marked this as pedantic and removed the changelog, but rest assured that we maintainers appreciate this a lot!

**Pull request merged**